snap/snapcraft.yaml

```diff
@@ -192,38 +176,122 @@ parts:
     after:
       - snapcraft-desktop-helpers
     plugin: nil
-    override-stage: |
+    override-build: |
+      snapcraftctl build
       rm -f $SNAPCRAFT_PART_INSTALL/usr/share/applications/python3.6.desktop
-      snapcraftctl stage
     stage-packages:
       - firefox
 
   gtk3-apps:
     plugin: nil
     stage-packages:
-      - gnome-sudoku
-      - gnome-mahjongg
       - gnome-chess
       - gnuchess
       - gnuchess-book
 
   native-apps:
     plugin: nil
+    override-build: |
+      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/share/applications/
+      cat > $SNAPCRAFT_PART_INSTALL/usr/share/applications/glmark2-wayland.desktop <<EOF
+      [Desktop Entry]
+      Name=GLMark2 (Wayland)
+      Exec=glmark2-es2-wayland
+      Type=Application
+      Icon=ubuntu-logo
+      Categories=Utility
+      EOF
+      cat > $SNAPCRAFT_PART_INSTALL/usr/share/applications/glmark2-x11.desktop <<EOF
+      [Desktop Entry]
+      Name=GLMark2 (X11)
+      Exec=glmark2-es2
+      Type=Application
+      Icon=ubuntu-logo
+      Categories=Utility
+      EOF
+      snapcraftctl build
     stage-packages:
+      - glmark2-es2
       - glmark2-es2-wayland
 
   kodi:
-    plugin: ppa
-    ppa: team-xbmc/ppa
-    override-stage: |
+    plugin: nil
+    override-build: |
+      snapcraftctl build
       if [ -e "$SNAPCRAFT_PART_INSTALL/usr/bin/kodi" ]; then
         sed --in-place s'!/usr!${SNAP}/usr!' $SNAPCRAFT_PART_INSTALL/usr/bin/kodi
       fi
       rm -f $SNAPCRAFT_PART_INSTALL/usr/share/applications/python2.7.desktop
-      snapcraftctl stage
     stage-packages:
-      - to amd64: [ kodi-wayland, kodi-repository-kodi, kodi-visualization-spectrum, samba-libs ]
-      - to  i386: [ kodi-wayland, kodi-repository-kodi, kodi-visualization-spectrum, samba-libs ]
+      - kodi
+      - kodi-wayland
+      - kodi-repository-kodi
+      - kodi-visualization-spectrum
+      - samba-libs
+      - samba-common-bin
+      - samba-common
+      - libfstrcmp0
+      - libpulse0
+      - libaudio2
+      - libsndio7.0
+      - va-driver-all
+
+  zoom:
+    after: [ zoom-deps ]
+    plugin: dump
+    source: https://zoom.us/client/latest/zoom_amd64.deb
+    override-build: |
+      # remove danging symlink
+      rm -f $SNAPCRAFT_PART_BUILD/usr/bin/zoom
+      echo "QT_QPA_PLATFORM=xcb \$SNAP/opt/zoom/ZoomLauncher" > $SNAPCRAFT_PART_BUILD/usr/bin/zoom
+      chmod +x $SNAPCRAFT_PART_BUILD/usr/bin/zoom
+      # Unfrig LD_LIBRARY_PATH
+      sed -i s/D_LIBRAR/XXXXXXXX/g opt/zoom/ZoomLauncher
+      snapcraftctl build
+    build-attributes: [keep-execstack]
+
+  zoom-deps:
```

Comment:
Why this split?

Reply:
It was convenient while iterating. Can probably be cleaned up now.